<launch>
  <!-- master DAVIS -->
  <node ns="davis_left" name="davis_ros_driver" pkg="davis_ros_driver" type="davis_ros_driver" output="screen">
    <param name="serial_number" type="str" value="00000306" />
    <rosparam command="load" file="$(find davis_ros_driver)/config/DAVIS240C.yaml" />
    <param name="master" value="True"/>
    <!-- timeout before resetting time -->
    <param name="reset_timestamps_delay" value="2.0"/>
  </node>

  <!-- slave DAVIS -->
  <node ns="davis_right" name="davis_ros_driver" pkg="davis_ros_driver" type="davis_ros_driver" output="screen">
    <param name="serial_number" type="str" value="00000311" />
    <rosparam command="load" file="$(find davis_ros_driver)/config/DAVIS240C.yaml" />
    <param name="master" value="False"/>
    <!-- get reset signal from master -->
    <remap from="reset_timestamps" to="/davis_left/reset_timestamps"/>
  </node>

  <arg name="stitch_find_mode" default="False"/>

  <!-- stitch -->
  <node name="davis_stereo_stitch" pkg="davis_stereo_stitch" type="davis_stereo_stitch" output="screen">
    <rosparam param="/homography">[0.385815138416276, -0.04302951951000493, 165.737557919644,
                                  -0.2033143592460488, 0.7718620876332584, 23.49481879941964,
                                  -0.001544193299986335, -0.0003502925835596213, 1]</rosparam>
    <rosparam param="/stitch_find_mode" subst_value="True">$(arg stitch_find_mode)</rosparam>
  </node>

  <!-- display -->
  <node name="image_view_stitch" pkg="rqt_image_view" type="rqt_image_view">
    <remap from="image" to="stitch_image"/>
  </node>

  <!-- <node name="image_view_left" pkg="rqt_image_view" type="rqt_image_view">
    <remap from="image" to="stitch_image"/>
  </node>

  <node name="image_view_right" pkg="rqt_image_view" type="rqt_image_view">
    <remap from="image" to="stitch_image"/>
  </node> -->

</launch>
